### PR TITLE
Added libvecpf library test [v2]

### PR DESCRIPTION
**Pull request by harish-24**
This test fetches latest library, compiles and runs tests in it.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
@PraveenPenguin Addressed comments in version 2.
[job.log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/729420/job.log.txt)

File: toolchain/libvecpf.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.error('%s is needed for the test to be run' % package)
+        tarball = self.fetch_asset(
+            "https://github.com/Libvecpf/libvecpf/archive/master.zip",
+            expire='7d')
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(self.srcdir, 'libvecpf-master')
+
+        build.make(self.srcdir, make='./configure')
```
Comment (narasimhan-v) [inline toolchain/libvecpf.py]: I guess this would not run for non root users. Please check

File: toolchain/libvecpf.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        fail_list = ['FAIL', 'XFAIL', 'ERROR']
+        failures = []
+        for failure in fail_list:
+            num_fails = re.compile(r"# %s:(.*)" %
+                                   failure).findall(results)[0].strip()
+            if int(num_fails):
+                failures.append({failure: num_fails})
+
+        if failures:
+            self.fail('Test failed with following:%s' % failures)
```
Comment (narasimhan-v) [inline toolchain/libvecpf.py]: Suggestion:

```
dict = {}
for line in results():
    dict[line.split()[1].strip(':')] = line.split()[2]

for key in ['FAIL', 'ERROR']:
    if int(dict[key]):
        print dict[key]
```

File: toolchain/libvecpf.py (hunk 1 above)
Comment (harish-24) [inline toolchain/libvecpf.py]: Checked. Working fine.

Comment (narasimhan-v) [inline toolchain/libvecpf.py]: Cool, Ok

File: toolchain/libvecpf.py (hunk 2 above)
Comment (harish-24) [inline toolchain/libvecpf.py]: Thought it would be unnecessary to do for all the lines of result.

Comment (narasimhan-v) [inline toolchain/libvecpf.py]: Since you are parsing all lines anyway, thought it would be better.
Up to you to decide on the approach.

Comment (PraveenPenguin) [inline toolchain/libvecpf.py]: @harish-24 are you going to adopt @narasimhan-v comment otherwise looks me good ?

Comment (harish-24) [inline toolchain/libvecpf.py]: @PraveenPenguin No, you can comment if anything is needed.

Comment (PraveenPenguin) [inline toolchain/libvecpf.py]: @harish-24 thanks looks good

**Pull request merged**